content/en/blog/_posts/2023-07-18-swap-beta1-graduation/index.md

```diff
@@ -0,0 +1,233 @@
+---
+
+**Author:** Itamar Holder (Red Hat)
+
+The 1.22 release [introduced Alpha support](/blog/2021/08/09/run-nodes-with-swap-alpha/)
+for configuring swap memory usage for Kubernetes workloads on a per-node basis.
+Now, in release 1.28, swap configuration graduates to Beta with many new improvements.
+
+Prior to version 1.22, Kubernetes did not provide support for swap memory on Linux systems.
+This was due to the inherent difficulty in guaranteeing and accounting for pod memory utilization
+when swap memory was involved. As a result, swap support was deemed out of scope in the initial
+design of Kubernetes, and the default behavior of a kubelet was to fail to start if swap memory
+was detected on a node.
+
+In version 1.22, the swap feature was initially introduced in its Alpha stage. This represented
+a significant advancement, providing users with the opportunity to experiment with the swap
+feature for the first time. However, as an Alpha version, it was not fully developed and had
+several issues, including inadequate support for cgroup v2, insufficient metrics and summary
+API statistics, inadequate testing, and more.
+
+Swap in Kubernetes has numerous [use cases](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/2400-node-swap/README.md#user-stories)
+for a wide range of users. As a result, significant effort has been put into making
+the Beta version of Swap more stable, robust, user-friendly, and to address issues regarding the
+Alpha version that are mentioned above. This represents a crucial step towards achieving the
+goal of fully supporting swap in Kubernetes in a controlled, reliable, and predictable manner.
+
+## How do I use it?
+The utilization of swap memory on a node where it has already been provisioned can be
+facilitated by the activation of the `NodeSwap` feature gate on the kubelet.
+Additionally, the failSwapOn configuration setting must be disabled, or the
+`--fail-swap-on` command line flag must be deactivated.
+
+It is possible to configure the `memorySwap.swapBehavior` option to define the manner in which a node utilizes swap memory. For instance,
+
+```yaml
+memorySwap:
+  swapBehavior: UnlimitedSwap
+```
+
+The available configuration options for `swapBehavior` are:
+- `UnlimitedSwap` (default): Kubernetes workloads can use as much swap memory as they
+  request, up to the system limit.
+- `LimitedSwap` (default): The utilization of swap memory by Kubernetes workloads is subject to limitations. Only Pods of Burstable QoS are permitted to employ swap.
+
+If configuration for `memorySwap` is not specified and the feature gate is
+enabled, by default the kubelet will apply the same behaviour as the
+`UnlimitedSwap` setting.
+
+Note that `NodeSwap` is supported for **cgroup v2** only. cgroup v1 is no longer supported.
+
+### Install a Swap-enabled cluster with kubeadm
+
+#### Before you begin
+
+You need to have installed the kubeadm tool, following the steps from Installing kubeadm.
```

Comment:
link to the guide for the install with kubeadm: https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm/

Reply:
done